src/lib/y2network/hostname_reader.rb

```diff
@@ -34,20 +37,23 @@ module Y2Network
   # @example Read hostname
   #   Y2Network::HostnameReader.new.hostname #=> "foo"
   class HostnameReader
+    attr_reader :install_inf_hostname
+
     include Yast::Logger
+    include Yast::Wicked
 
     # Returns the hostname
     #
-    # @note If the hostname cannot be determined, generate a random one.
+    # @note If the hostname cannot be determined, generate a random one
+    # in installed system (do not generate one in the installer).
     #
     # @return [String]
     def hostname
-      if (Yast::Mode.installation || Yast::Mode.autoinst) &&
-          Yast::FileUtils.Exists("/etc/install.inf")
-        fqdn = hostname_from_install_inf
+      if Yast::Mode.installation || Yast::Mode.autoinst
```

Comment:
will it work during update?

Reply:
no, but network is not modified during update at all. I mean ... hostname will not be updated in the updated system, but it will be displayed correctly e.g. in bash prompt (bcs it is preset by linuxrc)